src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueDebtAdaptor.sol

```diff
@@ -130,41 +127,44 @@ contract MorphoBlueDebtAdaptor is BaseAdaptor, MorphoBlueHelperLogic {
-     * @param _id identifier of a Morpho Blue market.
+     * @param market identifier of a Morpho Blue market.
      * @param _amountToBorrow the amount of `loanToken` to borrow on the specified MB market.
      */
-    function borrowFromMorphoBlue(Id _id, uint256 _amountToBorrow) public {
-        _validateMBMarket(_id);
-        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+    function borrowFromMorphoBlue(MarketParams memory market, uint256 _amountToBorrow) public {
+        _validateMBMarket(market);
+        Id id = MarketParamsLib.id(market);
         _borrowAsset(market, _amountToBorrow, address(this));
-        if (minimumHealthFactor > (_getHealthFactor(_id, market))) {
-            revert MorphoBlueDebtAdaptor__HealthFactorTooLow(_id);
+        if (minimumHealthFactor > (_getHealthFactor(id, market))) {
+            revert MorphoBlueDebtAdaptor__HealthFactorTooLow(market);
         }
     }
 
     /**
      * @notice Allows strategists to repay loan debt on Morph Blue Lending Market. Make sure to call addInterest() beforehand to ensure we are repaying what is required.
      * @dev Uses `_maxAvailable` helper function, see `BaseAdaptor.sol`.
-     * @param _id identifier of a Morpho Blue market.
+     * @param market identifier of a Morpho Blue market.
      * @param _debtTokenRepayAmount The amount of `loanToken` to repay.
      * NOTE - MorphoBlue reverts w/ underflow/overflow error if trying to repay with more than what cellar has. That said, we will accomodate for times that strategists tries to pass in type(uint256).max
      */
-    function repayMorphoBlueDebt(Id _id, uint256 _debtTokenRepayAmount) public {
-        _validateMBMarket(_id);
-        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+    function repayMorphoBlueDebt(MarketParams memory market, uint256 _debtTokenRepayAmount) public {
+        _validateMBMarket(market);
+        Id id = MarketParamsLib.id(market);
         _accrueInterest(market);
         ERC20 tokenToRepay = ERC20(market.loanToken);
         uint256 debtAmountToRepay = _maxAvailable(tokenToRepay, _debtTokenRepayAmount);
-        // using Morpho sharesLibrary we can calculate the sharesToRepay from the debtAmount
-        uint256 totalBorrowAssets = morphoBlue.market(_id).totalBorrowAssets;
-        uint256 totalBorrowShares = morphoBlue.market(_id).totalBorrowShares;
-        uint256 sharesToRepay = debtAmountToRepay.toSharesUp(totalBorrowAssets, totalBorrowShares);
-        uint256 sharesAccToMorphoBlue = morphoBlue.borrowShares(_id, address(this));
-        if (sharesAccToMorphoBlue < sharesToRepay) {
-            sharesToRepay = sharesAccToMorphoBlue;
-        }
         tokenToRepay.safeApprove(address(morphoBlue), debtAmountToRepay);
-        _repayAsset(market, sharesToRepay, address(this));
+
+        uint256 totalBorrowAssets = morphoBlue.totalBorrowAssets(id);
+        uint256 totalBorrowShares = morphoBlue.totalBorrowShares(id);
```

Comment:
same suggestion here you could use a more efficient helper

Reply:
Another good point, due to the timeline and the smaller gas savings considered here, we are comfortable leaving this.